### PR TITLE
statedb tree: optimize the time of the statedb commit phase

### PR DESCRIPTION
**Pull request by j75689**
### Description

1. Use `smt.MultiSet` to speed up update times
2. Add prometheus, pprof configuration

### Rationale

After testing, it is found that the `statedb` needs a lot of hash value calculation when updating leaf node data in the commit stage, and the calculation of the hash value will consume a lot of time, so the performance is improved by parallelizing the calculation

### Example
Test in `BlockSize: 300, TPS: 300`

Before: 
image

After: 
image

Before modification, `statedb` takes about 1.5-2.5s to write a block of data in the commit phase, and it only takes 400ms-800ms after optimization

Overall can get about `60%` optimization in the commit stage

### Changes

Notable changes:
* tree, stated

Note

/update-integration-keyfile

### CODE REVIEW

File: tree/treedb.go [hunk 1]
```diff
@@ -182,6 +185,30 @@ func SetNamespace(
 	return context.TreeDB
 }
 
+const (
+	treeRoutinePoolSize = 10240
```
Comment (guagualvcha) [inline tree/treedb.go:189]: Would this be too large?

**Review by forcodedancing — APPROVED** [review]

File: core/statedb/statedb.go [hunk 2]
```diff
@@ -523,19 +561,60 @@ func (s *StateDB) IntermediateRoot(cleanDirty bool) error {
...
+	err := taskPool.Submit(func() {
+		resultChan <- &treeUpdateResp{
+			role: accountTreeRole,
+			err:  s.AccountTree.MultiSet(pendingAccountItem...),
+		}
+	})
+	if err != nil {
+		return err
+	}
+	err = taskPool.Submit(func() {
```
Comment (guagualvcha) [inline core/statedb/statedb.go:588]: A bit concerned about the correctness of this part, please do fullnode catch up test and prover test with 10 more tps.

Comment (j75689) [reply core/statedb/statedb.go]: I synced with fullnode on the develop branch(no parallel) and it passed

File: core/blockchain.go [hunk 3]
```diff
@@ -208,11 +234,13 @@ func (bc *BlockChain) commitNewBlock(blockSize int, createdAt int64) (*block.Blo
 		}
 	}
 
+	start := time.Now()
 	// Intermediate state root.
-	err := s.IntermediateRoot(false)
+	err := s.IntermediateRoot(bc.taskPool, false)
```
[the remark this answers is not in the document]
Comment (j75689) [reply core/blockchain.go]: updated, change all to use `common.gopool`

File: tree/treedb.go (hunk 1 above)
Comment (j75689) [reply tree/treedb.go]: updated, and I made it configurable, default changed to `1024`

**Review by forcodedancing — APPROVED** [review]

**Review by guagualvcha — APPROVED** [review]

**Pull request merged**